proto/tribune_tts.proto

```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
 // Provides information to the synthesizer that specifies how to process the request.
 message SynthesizeConfig
 {
-    // Desired sampling frequency in hertz of synthesized audio.
+    // Desired sampling frequency in hertz of synthesized audio. The value 0 means use the default Synthesizer sampling rate.
     int32 sample_rate_hertz = 1;
+
+    // Requested format of the output audio stream.
+    OutputFormat output_format = 2;
+
+    // ISO 639-1 code of the language of text to be synthesized (may be overriden by SSML tags in request text).
+    string language = 3;
+
+    // Optional. Overrides the default synthesized speech parameters
+    AudioConfig audio_config = 4;
+
+    // Requested voice to be used to synthesize the text (may be overriden by SSML tags in request text).
+    Voice voice = 5;
+}
+
+// The requested format of the response audio data.
+enum OutputFormat
+{
+    PCM16 = 0;       // Uncompressed 16-bit signed integer samples.
+    OGG_VORBIS = 1;  // Ogg/Vorbis endoded data stream.
 }
 
-// `SynthesizeResponse` is the only message returned to the client by
-// `Synthesize`. A series of one or more `SynthesizeResponse`
-// messages are streamed back to the client.
+// Desfines the parameters of synthesized speech.
+message AudioConfig
+{
+    // The average speech pitch scaling factor. 1.0 means default voice pitch.
+    float pitch = 1;
+
+    // The speech range scaling factor. 1.0 means default voice range.
+    float range = 2;
+
+    // The speech rate (speed) scaling factor. 1.0 means default voice rate.
+    float rate = 3;
+
+    // The speech volume. 1.0 means default volume.
+    float volume = 4;
+}
+
+// Voice definition. If both name and gender is defined, the name takes precedence.
+// If only name is defined, the specified voice is used.
+// If only gender is defined, the default voice for the requested gender is used.
+message Voice
+{
+    // The name of the voice (empty string means default voice).
+    string name = 1;
```

Comment:
add `language_code` here?

Reply:
In SSML we have separate tags for switching between languages () and voices (), so perhaps placing a language directly in SynthesizeConfig is OK.